File: app/src/main/java/com/mifos/apache/fineract/data/services/IndividualLendingService.java
package com.mifos.apache.fineract.data.services;

import com.mifos.apache.fineract.data.models.payment.PlannedPaymentPage;
import com.mifos.apache.fineract.data.remote.EndPoints;

import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

/**
 * @author Rajan Maurya
 *         On 13/07/17.
 */
public interface IndividualLendingService {

    @GET(EndPoints.API_PORTFOLIO_PATH
            + "/individuallending/products/{productidentifier}"
            + "/cases/{caseidentifier}/plannedpayments")
    Observable<PlannedPaymentPage> getPaymentScheduleForCase(
            @Path("productidentifier") String productIdentifier,
            @Path("caseidentifier") String caseIdentifier,
            @Query("pageIndex") Integer pageIndex,
            @Query("size") Integer size,
            @Query("initialDisbursalDate") String initialDisbursalDate);
}

File: app/src/main/java/org/apache/fineract/data/services/IndividualLendingService.kt
package org.apache.fineract.data.services

import io.reactivex.Observable
import org.apache.fineract.data.models.payment.PlannedPaymentPage
import org.apache.fineract.data.remote.EndPoints
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

/**
 * Created by Ahmad Jawid Muhammadi on 2/6/20
 */
interface IndividualLendingService {

    @GET(EndPoints.API_PORTFOLIO_PATH
            + "/individuallending/products/{productidentifier}"
            + "/cases/{caseidentifier}/plannedpayments")
    fun getPaymentScheduleForCase(
            @Path("productidentifier") productIdentifier: String?,
            @Path("caseidentifier") caseIdentifier: String?,
            @Query("pageIndex") pageIndex: Int?,
            @Query("size") size: Int?,
            @Query("initialDisbursalDate") initialDisbursalDate: String?): Observable<PlannedPaymentPage?>?
}